File: python/7 kyu/413.py
# https://www.codewars.com/kata/5a8c1b06fd5777d4c00000dd/train/python

# Passed

def diagonal(matrix):
    principal_diagonal_values = [matrix[idx][idx] for idx in range(len(matrix))]
    secondary_diagonal_values = [matrix[idx][-idx - 1] for idx in range(len(matrix))]
    
    principal_score = sum(principal_diagonal_values)
    secondary_score = sum(secondary_diagonal_values)
    
    if principal_score > secondary_score:
        result = "Principal Diagonal win!"
    elif secondary_score > principal_score:
        result = "Secondary Diagonal win!"
    else:
        result = "Draw!"
        
    return result

output = diagonal([[2,2,2],
                   [4,2,6],
                   [1,8,2]])
print(output)

File: nodejs/7 kyu/413.js
// https://www.codewars.com/kata/5e4e8f5a72d9550032953717/train/javascript

// Passed

function* allRationals() {
  let baseFractions = [[1, 1]];
  yield baseFractions[0];

  while (true) {
    let nextBaseFractions = [];
    for (const baseFraction of baseFractions) {
      let [numerator, denominator] = baseFraction;
      let [fraction1, fraction2] = [
        [numerator, denominator + numerator],
        [numerator + denominator, denominator],
      ];
      nextBaseFractions.push(fraction1);
      nextBaseFractions.push(fraction2);

      yield fraction1;
      yield fraction2;
    }

    baseFractions = nextBaseFractions;
  }
}

const output = allRationals().next();
console.log(output);